src/System.Management.Automation/engine/Subsystem/SubsystemInfo.cs

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Management.Automation.Internal;
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// Class used to represent the metadata and state of a subsystem.
+    /// </summary>
+    public abstract class SubsystemInfo
+    {
+        #region "Metadata of a Subsystem (public)"
+
+        /// <summary>
+        /// The kind of a concrete subsystem.
+        /// </summary>
+        public SubsystemKind Kind { get; private set; }
+
+        /// <summary>
+        /// The type of a concrete subsystem.
+        /// </summary>
+        public Type SubsystemType { get; private set; }
+
+        /// <summary>
+        /// Indicate whether the subsystem allows to unregister an implementation.
+        /// </summary>
+        public bool AllowUnregistration { get; private set; }
+
+        /// <summary>
+        /// Indicate whether the subsystem allows to have multiple implementations registered.
+        /// </summary>
+        public bool AllowMultipleRegistration { get; private set; }
```

Comment:
I searched `Allow` in our code base and found tons of members/variables use just `allow/Allow`, but not `allows`. So I prefer to keep it as is.

Reply:
Yeah, I'd argue that most of those are biased by interfacing with PowerShell, so an `-AllowThis` parameter is in the imperative voice, as is an `[AllowThat]` attribute, since they are demands to something else.

But on an object with a private setter, the `AllowsMultipleRegistration` property isn't imperative but declarative. You can't set it to force it to allow multiple registrations, it's just letting you know whether it does.

That means it uses the indicative voice, which conjugates as `AllowsMultipleRegistration`, so that you can write things like:

```csharp
if (thisSubsystem.AllowsMultipleRegistration)
{
      // Register more things!
}
```

That sounds better than `if this subsystem allow multiple registration`. And because the setter is private there's never any code like:

```csharp
mySubsystem.AllowMultipleRegistration
```

The property is only ever used in tests.

Anyway, that's a very minor point and I'm happy with however you prefer!